src/FlipMove.js

```diff
@@ -69,7 +69,7 @@ class FlipMove extends Component {
     // can complete. Because we cannot mutate props, we make `state` the source
     // of truth.
     this.state = {
-      children: props.children.map(child => ({
+      children: Children.toArray(props.children).map(child => ({
```

Comment:
Sorry, still not sure what you mean by "passed back into the React virtual dom". Where would the modifications be coming from?

Let's say we have the following setup:

```jsx
class Whatever extends Component {
    render() {
        return (
            <FlipMove>
                <div key={1} />
                <div key={2} />
            </FlipMove>
        );
    }
}
```

If we inspect the React tree, it would look like this:

```jsx
<FlipMovePropConverter>
    <ComposedComponent> // AKA Flip Move
        <div key={1} />
        <div key={2} />
    </ComposedComponent>
</FlipMovePropConverter>
```

Let's say the parent component `Whatever` updates, so that two children are removed and a third is added:

```jsx
class Whatever extends Component {
    render() {
        return (
            <FlipMove>
                <div key={3} />
            </FlipMove>
        );
    }
}
```

`FlipMovePropConverter` will re-render, creating a new React element that has this new `children` prop of a single object. It's still going to run through the same process, though; on every `render`, `FlipMovePropConverter` calls `this.convertProps`, which will do the React.Children work.

Apologies for the back-and-forth, but I'd like to understand what you mean. As I understand it, React works by creating new elements on every render, and reconciling the differences between each virtual tree, doing DOM updates when the diff shows that a change is needed. I don't see how this process would undo the work of the prop converter, so if there _is_ some missing piece I'm not aware of, I'd like to learn about it :)